file_util/inc/leatherman/file_util/directory.hpp

```diff
@@ -0,0 +1,40 @@
+/**
+ * @file
+ * Declares utility functions for enumerating directories.
+ */
+#pragma once
+
+#include <string>
+#include <functional>
+
+namespace leatherman { namespace file_util {
+
+    /**
+     * Contains utility functions for enumerating directories.
+     */
+    struct __attribute__((visibility("default"))) directory
```

Comment:
In facter, these methods were bundled in a struct like this. Is that something we want to continue doing? Why was it done this way? (Still getting up to speed on the subtleties of C++)